clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -950,6 +950,45 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
   if (getLangOpts().SYCLIsHost && D && D->hasAttr<SYCLKernelAttr>())
     Fn->addFnAttr("sycl_kernel");
 
+  if (getLangOpts().SYCL && D && (D->hasAttr<SYCLIntelLoopFuseAttr>())) {
+    auto *A = D->getAttr<SYCLIntelLoopFuseAttr>();
```

Comment:
It's generally a bad code smell to do `hasAttr<>()` followed by `getAttr<>()` (I continue to look forward to when we can upgrade the code base to C++17 so we can use a better language feature for these cases.) I'd recommend:
```
if (getLangOpts().SYCL && D) {
  if (const auto *A = D->getAttr<...>()) {
    ...
  }
}
```